doc/development_process/api_lifecycle.rst

```diff
@@ -77,16 +77,70 @@ An API can be declared ``stable`` after fulfilling the following requirements:
 - Stable APIs can get backward compatible updates, bug fixes and security fixes
   at any time.
 
+Introducing incompatible changes
+================================
+
+A stable API, as described above strives to remain backwards-compatible through
+its life-cycle. There are however cases where fulfilling this objective prevents
+technical progress or is simply unfeasible without unreasonable burden on the
+maintenance of the API and its implementation(s).
+
+In order to restrict and control the introduction of a change that breaks the
+promise of backwards compatibility the following steps must be followed whenever
+such a change is considered necessary in order to accept it in the project:
+
+#. An RFC issue must be opened on GitHub with the following content:
+
+   .. code-block:: none
+
+      Title:     RFC: API Change: <subsystem>
+      Contents:  - Background information
+                 - Details on the proposed change
+                   - Function call changes
+                   - Device Tree changes (source and bindings)
+                   - Kconfig option changes
+                 - Impact to users of the API, including the steps required
+                   to adapt out-of-tree users of the API to the change
+
+#. A Pull Request must be opened on GitHub that includes the following:
+
+   - A title that matches the RFC issue
+   - The actual changes to the API:
+     - Changes to the API header file
+     - Changes to the API implementation(s)
+     - Changes to the relevant API documentation
+     - Changes to Device Tree source and bindings
+   - The changes required to adapt all in-tree users of the API to the change
+   - An entry in the "API Changes" section of the release notes for the next
+     upcoming release
+   - The labels ``API`` and ``API change``
+
+#. An email must be sent to the ``tsc``, ``users`` and ``devel`` mailing lists
```

Comment:
Could we put this info in RFC template so community could see it directly from issue reporting side?

Reply:
No, because this doesn't really apply to all RFCs. But maybe we want that to change in the future, it's a good idea.